runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -195,7 +195,7 @@ MM_IndexableObjectAllocationModel::layoutContiguousArraylet(MM_EnvironmentBase *
 		GC_SlotObject slotObject(env->getOmrVM(), arrayoidPtr);
 		slotObject.writeReferenceToSlot((omrobjectptr_t)leafOffset);
 		leafOffset += arrayletLeafSize;
-		arrayoidPtr += 1;
+		arrayoidPtr = GC_SlotObject::addToSlotAddress(arrayoidPtr, 1, compressed);
```

Comment:
We could have a non-static variant of this method and call for each iteration slotObject.addToSlotAddress(1, compressed);  while constructing slotObject only once before entering the loop. It would be slightly more readable to me, but I'm not pushing for it.